src/Adapter/MSTest.TestAdapter/TestingPlatformAdapter/MSTestBridgedTestFramework.cs

```diff
@@ -52,13 +54,50 @@ internal sealed class MSTestBridgedTestFramework : SynchronizedSingleSessionVSTe
 
         if (!MSTestDiscovererHelpers.InitializeDiscovery(request.AssemblyPaths, request.RunContext, request.FrameworkHandle))
         {
-            return Task.CompletedTask;
+            return;
         }
 
-#pragma warning disable CA1859 // Use concrete types when possible for improved performance
-        ITestExecutor testExecutor = new MSTestExecutor();
-#pragma warning restore CA1859 // Use concrete types when possible for improved performance
+        MSTestExecutor testExecutor = new();
         using (cancellationToken.Register(testExecutor.Cancel))
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                ApartmentState currentApartmentState = Thread.CurrentThread.GetApartmentState();
+                ApartmentState? requestedApartmentState = MSTestSettings.RunConfigurationSettings.ExecutionApartmentState;
```

Comment:
If this is set on Linux, I may want to emit some warning (both for runner and vstest).